flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/rules/physical/batch/DynamicPartitionPruningRuleTest.java

```diff
@@ -0,0 +1,371 @@
+    TestValuesCatalog catalog = new TestValuesCatalog("testCatalog", "test_database", true);
+
+    @Before
+    public void setup() {
+        catalog.open();
+        util.tableEnv().registerCatalog("testCatalog", catalog);
+        util.tableEnv().useCatalog("testCatalog");
+        TableConfig tableConfig = util.tableEnv().getConfig();
+        tableConfig.set(OptimizerConfigOptions.TABLE_OPTIMIZER_DYNAMIC_FILTERING_ENABLED, true);
+
+        // partition fact table.
+        String ddl1 =
+                "CREATE TABLE test_database.fact_part (\n"
+                        + "  id BIGINT,\n"
+                        + "  name STRING,\n"
+                        + "  amount BIGINT,\n"
+                        + "  price BIGINT,\n"
+                        + "  fact_date_sk BIGINT\n"
+                        + ") PARTITIONED BY (fact_date_sk)\n"
+                        + "WITH (\n"
+                        + " 'connector' = 'values',\n"
+                        + " 'partition-list' = 'fact_date_sk:1990;fact_date_sk:1991;fact_date_sk:1992',\n"
+                        + " 'dynamic-filtering-fields' = 'fact_date_sk;amount',\n"
+                        + " 'bounded' = 'true'\n"
+                        + ")";
+        util.tableEnv().executeSql(ddl1);
+
+        // dim table.
+        String ddl2 =
+                "CREATE TABLE test_database.dim (\n"
+                        + "  id BIGINT,\n"
+                        + "  male BOOLEAN,\n"
+                        + "  amount BIGINT,\n"
+                        + "  price BIGINT,\n"
+                        + "  dim_date_sk BIGINT\n"
+                        + ") WITH (\n"
+                        + " 'connector' = 'values',\n"
+                        + " 'bounded' = 'true'\n"
+                        + ")";
+        util.tableEnv().executeSql(ddl2);
+    }
+
+    @Test
+    public void testDimTableFilteringFieldsNotInJoinKey() {
+        // fact_part.id not in dynamic-filtering-fields, so dynamic partition pruning will not
+        // succeed.
+        String query =
+                "Select * from dim, fact_part where fact_part.id = dim.id and dim.price < 500";
+        util.verifyRelPlan(query);
+    }
+
+    @Test
+    public void testDimTableWithoutFilter() {
+        // If dim side without filters, dynamic partition pruning will not succeed.
+        String query =
+                "Select * from dim, fact_part where fact_part.fact_date_sk = dim.dim_date_sk"
+                        + " and fact_part.price > 100";
+        util.verifyRelPlan(query);
+    }
+
+    @Test
+    public void testDimTableWithUnsuitableFilter() {
+        // For filters in dim table side, they need to filter enough partitions. Like NOT NULL or IS
+        // TRUE will
+        // not succeed for dynamic partition pruning.
+        String query =
+                "Select * from dim, fact_part where fact_part.fact_date_sk = dim.dim_date_sk and male = true";
```

Comment:
There is 50% percentage for `male = true` or `male is true`

Reply:
> There is 50% percentage for `male = true` or `male is true`

done! change to `IS NOT NULL`.